libraries/src/Router/Route.php

```diff
@@ -121,7 +121,7 @@ public static function _($url, $xhtml = true, $tls = self::TLS_IGNORE, $absolute
 	public static function link($client, $url, $xhtml = true, $tls = self::TLS_IGNORE, $absolute = false)
 	{
 		// If we cannot process this $url exit early.
-		if (!is_array($url) && (strpos($url, '&') !== 0) && (strpos($url, 'index.php') !== 0))
+		if (!is_array($url) && (strpos((string) $url, '&') !== 0) && (strpos((string) $url, 'index.php') !== 0))
```

Comment:
```suggestion
		if (is_null($url) || (!is_array($url) && (strpos($url, '&') !== 0) && (strpos($url, 'index.php') !== 0)))
```

Passing null in the first place does not sound to be a good thing in the first place right? If anything we should not try to hide that issue by casting anything to string right? Anything that is passed should be a string in the first place already by the signature.